pokerole_irl/worlds_app/views.py

```diff
@@ -0,0 +1,59 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.shortcuts import redirect
+from django.views.generic import CreateView, ListView, DetailView
+from .models import Setting, Character
+
+# Create your views here.
+
+class SettingListView(LoginRequiredMixin, ListView):
+    template_name = "users/setting_list.html"
+    model = Setting
+    context_object_name = "settings"
+    paginate_by = 20
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        return queryset.filter(members=self.request.user)
+
+class SettingView(LoginRequiredMixin, DetailView):
+    template_name = "users/setting.html"
+    model = Setting
+    context_object_name = "setting"
+
+    def get_context_data(self, **kwargs):
+        # Call the base implementation first to get a context
+        context = super(SettingView, self).get_context_data(**kwargs)
+        # Add in a QuerySet of all charactes in the setting
+        context['character_list'] = Character.objects.filter(setting=self.object.id)
+        return context
+
+
+
+class SettingCreateView(LoginRequiredMixin, CreateView):
+    template_name = "users/setting_create.html"
+    model = Setting
+    fields = ("name", "description", "members")
+
+    def form_valid(self, form):
+        prod = form.save(commit=False)
+        prod.owner = self.request.user
+        prod.save()
+        form.save_m2m()
+        return redirect("setting", pk=prod.id)
```

Comment:
Instead of doing this, use the View's object attribute:

```
self.object = form.save(commit=False)
self.object.owner = self.request.user
return super().form_valid(form)
```

Since the super().form_valid(form) will save the object, then handle redirects automatically. You probably still need the `form.save_m2m()`